homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js

```diff
@@ -0,0 +1,66 @@
+// 1 task
+const exampleArray = [10, -10, 10, -10, 10];
+function sumArray(arr) {
+    const resultArray = [];
+    arr.reduce((previousSum, currentNumber) => {
+        const value = previousSum + currentNumber;
+        resultArray.push(value);
+        return value;
+    }, 0);
+    console.log(resultArray);
+}
+sumArray(exampleArray);
+// 2 task
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+function repeatsInObj(array) {
+    const obj = array.reduce((classList, className) => {
+        const cl = classList;
+        const cn = className;
+        if (cn in cl) {
+            cl[cn]++;
+        } else {
+            cl[cn] = 1;
+        }
+        return classList;
+    }, {});
+    console.log(obj);
+}
+repeatsInObj(classes);
+// 3 task
+const objForCopy = { b: 'f', d: { e: 'f' } };
+function objCopy(obj) {
+    if (typeof obj === 'object') {
+        return Object.keys(obj)
+            .map(k => ({ [k]: objCopy(obj[k]) }))
+            .reduce((a, c) => Object.assign(a, c), {});
```

Comment:
idea is right, but make it more readable, you just need to look at each object property and do action according this property data type

keep it simple please